### PR TITLE
fix the bug when the expiration dates can't be removed from link perm…

### PR DESCRIPTION
**Pull request by 2403905**
…issions

## Related Issue

- Fixes [https://github.com/owncloud/ocis/issues/8405](https://github.com/owncloud/ocis/issues/8405)

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [X] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/graph/pkg/service/v0/links.go [hunk 1]
```diff
@@ -202,7 +202,7 @@ func (g Graph) libreGraphPermissionFromCS3PublicShare(createdLink *link.PublicSh
 }
 
 func parseAndFillUpTime(t *time.Time) *types.Timestamp {
```
Comment (micbar) [inline services/graph/pkg/service/v0/links.go:204]: @rhafer @fschade This should also apply to the normal sharing link expiration.

Comment (rhafer) [reply services/graph/pkg/service/v0/links.go:204]: @micbar So we want to limit the granularity of share expiration to a day? Didn't know that. Shouldn't that be done in the CS3 shareprovider then?

Comment (fschade) [reply services/graph/pkg/service/v0/links.go:204]: @micbar needs discussion, the isZero check is valid and needed... but the rounding seems to be something we should talk about, the api allows setting not just the day but instead also allows to define the exact time ... so no rounding is needed, but we could be wrong, you have some minutes to talk about it?

**Review by rhafer — APPROVED** [review]
Regardless of the discussion about the granularity this fix is valid. So let's merge it an worry about the granularity in a separate issue.

**Pull request merged**